atd-vzv/src/views/summary/SummaryView.js

```diff
@@ -26,54 +30,96 @@ import {
 
 const SummaryView = () => {
   const [fatalities, setFatalities] = useState(null);
+  const [yearsOfLifeLost, setYearsOfLifeLost] = useState(null);
   const [seriousInjuries, setSeriousInjuries] = useState(null);
   const [totalCrashes, setTotalCrashes] = useState(null);
-  const [yearsOfLifeLost, setYearsOfLifeLost] = useState(null);
 
   useEffect(() => {
-    axios.get(fatalitiesYTDUrl).then(res => {
+    let fatalitiesUrl, yearsOfLifeLostUrl, seriousInjuriesUrl, totalCrashesUrl;
+
+    if (thisMonth > "01") {
+      fatalitiesUrl = thisYearFatalitiesUrl;
+      yearsOfLifeLostUrl = thisYearYearsOfLifeLostUrl;
+      seriousInjuriesUrl = thisYearSeriousInjuriesUrl;
+      totalCrashesUrl = thisYearTotalCrashesUrl;
+    } else {
+      fatalitiesUrl = previousYearFatalitiesUrl;
+      yearsOfLifeLostUrl = previousYearYearsOfLifeLostUrl;
+      seriousInjuriesUrl = previousYearSeriousInjuriesUrl;
+      totalCrashesUrl = previousYearTotalCrashesUrl;
+    }
+
+    axios.get(fatalitiesUrl).then(res => {
       setFatalities(calculateTotalFatalities(res.data));
     });
 
-    axios.get(seriousInjuriesYTDUrl).then(res => {
-      setSeriousInjuries(calculateTotalInjuries(res.data));
+    axios.get(yearsOfLifeLostUrl).then(res => {
+      setYearsOfLifeLost(getYearsOfLifeLost(res.data));
     });
 
-    axios.get(totalCrashesYTDUrl).then(res => {
-      setTotalCrashes(calculateTotalCrashes(res.data));
+    axios.get(seriousInjuriesUrl).then(res => {
+      setSeriousInjuries(calculateTotalInjuries(res.data));
     });
 
-    axios.get(yearsOfLifeLostYTDUrl).then(res => {
-      setYearsOfLifeLost(getYearsOfLifeLost(res.data));
+    axios.get(totalCrashesUrl).then(res => {
+      setTotalCrashes(calculateTotalCrashes(res.data));
     });
   }, []);
 
-  const summaryWidgetsConfig = [
-    {
-      title: `Fatalities in ${thisYear}`,
-      total: fatalities,
-      icon: faHeartbeat,
-      color: colors.danger
-    },
-    {
-      title: `Serious Injuries in ${thisYear}`,
-      total: seriousInjuries,
-      icon: faMedkit,
-      color: colors.warning
-    },
-    {
-      title: `Total Crashes in ${thisYear}`,
-      total: totalCrashes,
-      icon: faCar,
-      color: colors.success
-    },
-    {
-      title: `Years of Life Lost in ${thisYear}`,
-      total: yearsOfLifeLost,
-      icon: faHourglass,
-      color: colors.info
-    }
-  ];
+  const summaryWidgetsConfig =
```

Comment:
I think you could make the code more concise here by focusing on the specific substring that needs to change based on your conditional instead of having two separate big array of objects defined.

```js
let displayYear = thisMonth > "01" ? thisYear : lastYear;

const summaryWidgetsConfig = [
      {
            title: `Fatalities in ${displayYear}`,
            total: fatalities,
            icon: faHeartbeat,
            color: colors.danger
          },
          {
            title: `Years of Life Lost in ${displayYear}`,
            total: yearsOfLifeLost,
            icon: faHourglass,
            color: colors.info
          },
          {
            title: `Serious Injuries in ${displayYear}`,
            total: seriousInjuries,
            icon: faMedkit,
            color: colors.warning
          },
          {
            title: `Total Crashes in ${displayYear}`,
            total: totalCrashes,
            icon: faCar,
            color: colors.success
          }
]

```

Reply:
Yes! This is what I was ultimately going for. Thanks!